doc/sphinxext/dropdown_anchors.py

```diff
@@ -0,0 +1,67 @@
+import re
+
+from docutils import nodes
+from sphinx.transforms.post_transforms import SphinxPostTransform
+from sphinx_design.dropdown import dropdown_main, dropdown_title
+
+
+class DropdownAnchorAdder(SphinxPostTransform):
+    """Insert anchor links to the sphinx-design dropdowns.
+
+    Some of the dropdowns were originally headers that had automatic anchors, so we
+    need to make sure that the old anchors still work. See the original implementation
+    (in JS): https://github.com/scikit-learn/scikit-learn/pull/27409
```

Comment:
I think we never really used the anchors. Maybe we can simplify the code under this assumption and test _a posteriori_ with a link checker.

Reply:
Well in fact I'm not even sure if this comment is the actual intention of `details-permalink.js` because it is not me who implemented it.

> I just noticed sphinx already has an anchor convention for repeated titles, for instance see https://scikit-learn.org/stable/modules/ensemble.html#id27, so maybe we can keep their convention or something similar for the sake of compatibility. WDYT?

I'm not sure about this, maybe it should be up to maintainers to decide. From the implementation side, I need to find out a way to check all IDs that sphinx added automatically to avoid conflicts. And after second thought the current implementation may suffer from duplicated IDs as well because we only check our own IDs but not those added by sphinx or other extensions. This is a tricky one and I may need further investigation as well as comments from maintainers who I've pinged in the top comment.

Reply:
I don't think `sphinx` is able to detect duplicated IDs added like this? For instance if I tried removing the `anchor_id_counters` and got no warnings.